Sources/MapboxCoreNavigation/EHorizon/RoadObjectMatcher.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+import MapboxNavigationNative
+
+/**
+ Provides methods for road object matching.
+
+ Matching results are delivered asynchronously via a delegate.
+ In case of error (if there are no tiles in the cache, decoding failed, etc.) the object won't be matched.
+ */
+final public class RoadObjectMatcher {
+
+    /// Road object matcher delegate.
+    public weak var delegate: RoadObjectMatcherDelegate? {
+        didSet {
+            if delegate != nil {
+                native.setListenerFor(self)
+            } else {
+                native.setListenerFor(nil)
+            }
+        }
+    }
+
+    /**
+     Matches given OpenLR object to the graph.
+
+     - parameter location: OpenLR location of the road object, encoded in a base64 string.
+     - parameter standard: Standard used to encode OpenLR location.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchOpenLR(location: String, standard: OpenLRStandard, identifier: RoadObjectIdentifier) {
+        native.matchOpenLR(forBase64Encoded: location, standard: MapboxNavigationNative.OpenLRStandard(standard), id: identifier)
+    }
+
+    /**
+     Matches given polyline to the graph.
+     Polyline should define a valid path on the graph,
+     i.e. it should be possible to drive this path according to traffic rules.
+
+     - parameter polyline: Polyline representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchPolyline(_ polyline: [CLLocation], identifier: RoadObjectIdentifier) {
+        native.matchPolyline(forPolyline: polyline, id: identifier)
+    }
+
+    /**
+     Matches a given polygon to the graph.
+     "Matching" here means we try to find all intersections of the polygon with the road graph
+     and track distances to those intersections as distance to the polygon.
+
+     - parameter polygon: Polygon representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchPolygon(_ polygon: [CLLocation], identifier: RoadObjectIdentifier) {
+        native.matchPolygon(forPolygon: polygon, id: identifier)
+    }
+
+    /**
+     Matches given gantry (i.e. polyline orthogonal to the road) to the graph.
+     "Matching" here means we try to find all intersections of the gantry with the road graph
+     and track distances to those intersections as distance to the gantry.
+
+     - parameter gantry: Gantry representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchGantry(_ gantry: [CLLocation], identifier: RoadObjectIdentifier) {
```

Comment:
This method should be called `match(gantry:identifier:)` and accept a `Turf.LineString`.

Reply:
Ah, OK, does it need to be an array of `CLLocation`s though? That would imply that things like timestamps and headings are respected. I’m pretty sure the navigator just uses CLLocation because that’s what the automated Objective-C header generation tool uses. An array of `CLLocationCoordinate2D` or a `Turf.MultiPoint` would be more intuitive.

Reply:
`Turf.MultiPoint` is used here.